libs/api/metadata-converter/src/lib/xml-utils.ts

```diff
@@ -314,7 +314,19 @@ export function createElement(
   return () => new XmlElement(name, {}, [])
 }
 
-export function addAttribute(
+export function createNestedElement(
```

Comment:
Maybe some added tests for the two new functions here would make sense?

Reply:
I haven't tested them because I considered that they were sufficiently covered by all the other tests in this module. But I added some comments to clarify what they do and hopefully simplified a bit the utils, because it might have been hard to understand from outside.